src/commands/test.js

```diff
@@ -1,525 +1,63 @@
-    try {
-      await startTestEnvironment(tempdir)
-    } catch (e) {
-      print.error(e)
-      process.exitCode = 1
-      return
-    }
-
-    // Wait for test environment to come up
-    try {
-      await waitForTestEnvironment({
-        skipWaitForEthereum,
-        skipWaitForIpfs,
-        skipWaitForPostgres,
-	timeout,
-      })
-    } catch (e) {
-      await stopTestEnvironment(tempdir)
-      process.exitCode = 1
-      return
-    }
-
-    // Bring up Graph Node separately, if a standalone node is used
-    let nodeProcess
-    let nodeOutputChunks = []
-    if (standaloneNode) {
-      try {
-        nodeProcess = await startGraphNode(
-          standaloneNode,
-          standaloneNodeArgs,
-          nodeOutputChunks,
-        )
-      } catch (e) {
-        toolbox.print.error('')
-        toolbox.print.error('  Graph Node')
-        toolbox.print.error('  ----------')
-        toolbox.print.error(
-          indent('  ', Buffer.concat(nodeOutputChunks).toString('utf-8')),
-        )
-        toolbox.print.error('')
-        await stopTestEnvironment(tempdir)
-        process.exitCode = 1
-        return
-      }
-    }
-
-    // Wait for Graph Node to come up
-    try {
-      await waitForGraphNode(timeout)
-    } catch (e) {
-      toolbox.print.error('')
-      toolbox.print.error('  Graph Node')
-      toolbox.print.error('  ----------')
-      toolbox.print.error(
-        indent(
-          '  ',
-          await collectGraphNodeLogs(tempdir, standaloneNode, nodeOutputChunks),
-        ),
-      )
-      toolbox.print.error('')
-      await stopTestEnvironment(tempdir)
-      process.exitCode = 1
-      return
-    }
-
-    // Run tests
-    let result = await runTests(testCommand)
-
-    // Bring down Graph Node, if a standalone node is used
-    if (nodeProcess) {
-      try {
-        await stopGraphNode(nodeProcess)
-      } catch (e) {
-        // do nothing (the spinner already logs the problem)
-      }
-    }
-
-    if (result.exitCode == 0) {
-      toolbox.print.success('✔ Tests passed')
-    } else {
-      toolbox.print.error('✖ Tests failed')
-    }
-
-    // Capture logs
-    nodeLogs =
-      nodeLogs || result.exitCode !== 0
-        ? await collectGraphNodeLogs(tempdir, standaloneNode, nodeOutputChunks)
-        : undefined
-    ethereumLogs = ethereumLogs ? await collectEthereumLogs(tempdir) : undefined
-
-    // Bring down the test environment
-    try {
-      await stopTestEnvironment(tempdir)
-    } catch (e) {
-      // do nothing (the spinner already logs the problem)
-    }
-
-    if (nodeLogs) {
-      toolbox.print.info('')
-      toolbox.print.info('  Graph node')
-      toolbox.print.info('  ----------')
-      toolbox.print.info('')
-      toolbox.print.info(indent('  ', nodeLogs))
+    const platform = getPlatform();
+    if (!version) {
+      let xmlHttp = new XMLHttpRequest();
+      xmlHttp.open('GET', 'https://api.github.com/repos/LimeChain/matchstick/releases/latest', false);
```

Comment:
I think it's a good idea to lock the version of `matchstick`.

Nowadays for example we have `graph-ts` locked to a version because sometimes we do releases with breaking changes.

Reply:
We talked with Leo and Adam about this problem. If we want to lock the version of graph-cli to matchstick this would mean that we would have to release a new version of graph-cli every time we release a new one for matchstick (keep in mind that we are in something like an open beta right now so severe bugs are somewhat frequent) and that's why we decided it would be better to always download the latest version so that we can freely add bug fixes and new features whenever necessary while also having the version flag available if someone wants to download an older binary and in our readme file we will have a table of compatible versions.

Reply:
Ok!